src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -276,9 +276,21 @@ private bool HasMultipleReplicas(ResourceViewModel resource)
         return false;
     }
 
-    private string? GetRowClass(ResourceViewModel resource)
+    private string GetRowClass(ResourceViewModel resource)
         => resource == SelectedResource ? "selected-row resource-row" : "resource-row";
 
+    private string GetGridTemplateColumns()
+    {
+        if (ViewportInformation.IsDesktop)
+        {
+            return HasResourcesWithCommands ? "1fr 1.5fr 1.25fr 1.5fr 2.5fr 2.5fr 1fr 1fr 1fr" : "1fr 1.5fr 1.25fr 1.5fr 2.5fr 2.5fr 1fr 1fr";
+        }
+        else
+        {
+            return HasResourcesWithCommands ? "1fr 1.5fr 1.25fr 1fr 1fr" : "1fr 1.5fr 1.25fr 1fr";
+        }
```

Comment:
This is pretty confusing. Also, I think it will be very easy to go out of date with the columns that are shown. I think a more structured approach is needed.

What about adding a type called `GridColumnManager`. It would be initialized with definitions for the grid columns, e.g.

```cs
public void OnInitialize()
{
    _manager = new GridColumnManager([
        new GridColumn(name: "Type", desktopWidth: "1fr", notDesktopWidth: "1fr"),
        new GridColumn(name: "Source", desktopWidth: "2.5fr"),
        new GridColumn(name: "Commands", desktopWidth: "1fr", notDesktopWidth: "1fr", visible: () => HasResourcesWithCommands),
    ]);
}

public void OnParametersSet()
{
    _manager.SetViewport(ViewportInformation);
}

// This method in the manager would loop over the grid column definitions and concat all that are visible.
private string GetGridTemplateColumns() => _manager.GetGridTemplateColumns();
```

Then in razor, we ask the manager whether a column is visible or not.

```razor
// LogLevel here could be a constant to share between definitions and razor
@if (_manager.IsColumnVisible("LogLevel"))
{
    // hide this on mobile, info is available in detail panel + row background
    // helps to indicate severity for sighted users
    <TemplateColumn Title="@Loc[nameof(Dashboard.Resources.StructuredLogs.StructuredLogsLevelColumnHeader)]">
        <LogLevelColumnDisplay LogEntry="@context" />
    </TemplateColumn>
}
```

I think this makes logic centralized and understandable.

Reply:
I have now added a `GridColumnManager` to do this